consensus/istanbul/backend/engine_test.go

```diff
@@ -812,3 +821,184 @@ func TestSnapshot(t *testing.T) {
+				{vote{"governance.governancemode", "single"}, 3},
+				{vote{"governance.governancemode", "single"}, 4},
+				{vote{"governance.governancemode", "single"}, 5},
+				{vote{"governance.governancemode", "none"}, 6},
+				{vote{"governance.governancemode", "none"}, 7},
+				{vote{"governance.governancemode", "none"}, 8},
+				{vote{"governance.governancemode", "single"}, 9},
+				{vote{"governance.governancemode", "single"}, 10},
+				{vote{"governance.governancemode", "single"}, 11},
+				{vote{"governance.governancemode", "none"}, 12},
+				{vote{"governance.governancemode", "none"}, 13},
+				{vote{"governance.governancemode", "none"}, 14},
+				{vote{"governance.governancemode", "none"}, 0}, // check on current
+
+				// committee size for all blocks
+				{vote{"istanbul.committeesize", uint64(21)}, 1},
+				{vote{"istanbul.committeesize", uint64(21)}, 2},
+				{vote{"istanbul.committeesize", uint64(21)}, 3},
+				{vote{"istanbul.committeesize", uint64(21)}, 4},
+				{vote{"istanbul.committeesize", uint64(21)}, 5},
+				{vote{"istanbul.committeesize", uint64(4)}, 6},
+				{vote{"istanbul.committeesize", uint64(4)}, 7},
+				{vote{"istanbul.committeesize", uint64(4)}, 8},
+				{vote{"istanbul.committeesize", uint64(22)}, 9},
+				{vote{"istanbul.committeesize", uint64(22)}, 10},
+				{vote{"istanbul.committeesize", uint64(22)}, 11},
+				{vote{"istanbul.committeesize", uint64(22)}, 12},
+				{vote{"istanbul.committeesize", uint64(22)}, 13},
+				{vote{"istanbul.committeesize", uint64(22)}, 14},
+				{vote{"istanbul.committeesize", uint64(22)}, 0}, // check on current
+
+				// unitprice for all blocks
+				{vote{"governance.unitprice", uint64(1)}, 1},
+				{vote{"governance.unitprice", uint64(1)}, 2},
+				{vote{"governance.unitprice", uint64(1)}, 3},
+				{vote{"governance.unitprice", uint64(1)}, 4},
+				{vote{"governance.unitprice", uint64(1)}, 5},
+				{vote{"governance.unitprice", uint64(1)}, 6},
+				{vote{"governance.unitprice", uint64(1)}, 7},
+				{vote{"governance.unitprice", uint64(1)}, 8},
+				{vote{"governance.unitprice", uint64(2000000)}, 9},
+				{vote{"governance.unitprice", uint64(2000000)}, 10},
+				{vote{"governance.unitprice", uint64(2000000)}, 11},
+				{vote{"governance.unitprice", uint64(2)}, 12},
+				{vote{"governance.unitprice", uint64(2)}, 13},
+				{vote{"governance.unitprice", uint64(2)}, 14},
+				{vote{"governance.unitprice", uint64(2)}, 0}, // check on current
+			},
+		},
+	}
+
+	var configItems []interface{}
+	configItems = append(configItems, proposerPolicy(params.WeightedRandom))
+	configItems = append(configItems, epoch(3))
+	configItems = append(configItems, governanceMode("single"))
+	configItems = append(configItems, blockPeriod(0)) // set block period to 0 to prevent creating future block
+	for _, tc := range testcases {
+		chain, engine := newBlockChain(1, configItems...)
+
+		// test initial governance items
+		assert.Equal(t, uint64(3), engine.governance.Epoch())
```

Comment:
@yoomee1313 Right. There is the same problem for the epoch since it uses the current epoch only. I think it is not possible to add the test at this moment. I will make an issue for this.